### PR TITLE
Add label and annotation support

### PR DESCRIPTION
**Pull request by cjellick**
This change adds support for adding labels and annotations to the
resources created by acorn. The labels and annotations can be specified
in the Acornfile or on the command line and can be scoped to apply to
all resources, resources of a specific type (ie containers or volumes),
or resources with a specific name.

Addresses:
- https://github.com/acorn-io/acorn/issues/491
- https://github.com/acorn-io/acorn/issues/571

### CODE REVIEW

File: schema/v1/app.cue [hunk 1]
```diff
@@ -15,6 +15,10 @@ package v1
 
 #EnvVars: *[...string] | {[string]: string}
 
+#Labels: string
```
Comment (cjellick) [inline schema/v1/app.cue:18]: not necessary. just use string

File: schema/v1/app.cue [hunk 2]
```diff
@@ -94,6 +98,8 @@ package v1
 	ports:                          #PortSingle | *[...#Port] | #PortMap
 	[=~"probes|probe"]:             #Probes
 	[=~"depends[oO]n|depends_on"]:  string | *[...string]
+	labels:                         [string]: #Labels
```
Comment (cjellick) [inline schema/v1/app.cue:101]: move out of base and into job and container

Comment (cjellick) [reply schema/v1/app.cue:101]: dont want it on sidecar

File: pkg/ports/ports.go [hunk 3]
```diff
@@ -96,11 +95,29 @@ func ToContainerServices(app *v1.AppInstance, publish bool, namespace string, po
 		if publish {
 			extraLabels = append(extraLabels, labels.AcornServicePublish, "true")
 		}
+
+		labelMap := labels.Managed(app, extraLabels...)
+		anns := make(map[string]string)
+		// This is complicated, but we need to do this because a service can be selecting multiple containers, if both
+		// containers have a port with the same serviceName. So, this logic finds all the containers a service is selecting and
+		// gathers the labels/annotations from them.
+		if ports, ok := portSet.Services[serviceName]; ok {
```
Comment (cjellick) [inline pkg/ports/ports.go:104]: this is likely missing support for services that target nested acorns

Comment (cjellick) [reply pkg/ports/ports.go:104]: resolved. opened issue #641

File: schema/v1/app.cue (hunk 2 above)
Comment (cjellick) [reply schema/v1/app.cue]: resolved. moved these out of container base

File: schema/v1/app.cue (hunk 1 above)
Comment (cjellick) [reply schema/v1/app.cue]: fixed

**Review by ibuildthecloud — APPROVED** [review]

**Pull request merged**